src/lib/coda_base/pending_coinbase.ml

```diff
@@ -767,29 +1004,30 @@ struct
     let%bind index = find_index t key in
     get_stack t index
 
-  let add_coinbase t ~coinbase ~is_new_stack =
+  let update_stack' t ~(f : Stack.t -> Stack.t) ~is_new_stack =
     let open Or_error.Let_syntax in
-    let%bind key = latest_stack_id t ~is_new_stack in
+    let key = latest_stack_id t ~is_new_stack in
     let%bind stack_index = find_index t key in
     let%bind stack_before = get_stack t stack_index in
-    let stack_after = Stack.push stack_before coinbase in
-    let%bind t' = incr_index t ~is_new_stack in
+    let stack_after = f stack_before in
```

Comment:
Should this not still use `incr_index` before `set_stack`?

Reply:
`incr_index` is called inside `set_stack`